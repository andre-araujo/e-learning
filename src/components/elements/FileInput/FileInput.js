import { Component } from 'react';
import { string, func } from 'prop-types';
import {
  FaFile,
} from 'react-icons/fa';

import Text from '../Text';

import {
  Label,
} from './FileInput.styles';

class FileInput extends Component {
  state = {
    fileName: 'Clique para adicionar uma mÃ­dia',
  }

  onChange = (e) => {
    const {
      onChange,
    } = this.props;

    this.setState({
      fileName: e.target.files[0].name,
    });

    if (onChange) onChange(e.target.files[0]);
  }

  render() {
    const {
      label,
      name,
      ...inputProps
    } = this.props;
    const {
      fileName,
    } = this.state;

    return (
      <Label>
        <input {...inputProps} name={name} type="file" onChange={this.onChange} />
        <FaFile />
        <Text.Small margin="10px">
          {fileName}
        </Text.Small>
      </Label>
    );
  }
}

FileInput.defaultProps = {
  label: null,
  onChange: null,
};

FileInput.propTypes = {
  name: string.isRequired,
  label: string,
  onChange: func,
};

export default FileInput;
