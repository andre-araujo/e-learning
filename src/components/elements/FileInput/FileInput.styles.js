import styled from 'styled-components';

export const Label = styled.label`
  height: 100px;
  width: 100%;
  margin: 10px 0;
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  border: 1px dashed #aaa;
  cursor: pointer;
  flex-direction: column;
  text-align: center;

  input {
    display: none;
  }
`;
