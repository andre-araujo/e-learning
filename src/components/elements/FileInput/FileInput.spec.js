import React from 'react';
import { shallow, mount } from 'enzyme';

import FileInput from './FileInput';

describe('FileInput component', () => {
  it('should match snapshot', () => {
    const onChange = jest.fn();
    expect(shallow(<FileInput name="file" onChange={onChange} />)).toMatchSnapshot();
  });

  it('should should call onChange prop', () => {
    const onChange = jest.fn();
    const fileInput = shallow(<FileInput name="file" onChange={onChange} />);
    fileInput.find('input').simulate('change', { target: { files: [{ name: 'sample' }] } });

    expect(onChange).toHaveBeenCalledWith({ name: 'sample' });
  });

  it('should set file state on change', () => {
    const fileInput = mount(<FileInput name="file" />);
    fileInput.find('input').simulate('change', { target: { files: [{ name: 'sample' }] } });

    expect(fileInput.state('fileName')).toEqual('sample');
  });
});
