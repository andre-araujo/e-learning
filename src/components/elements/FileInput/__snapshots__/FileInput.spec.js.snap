// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput component should match snapshot 1`] = `
<styled.label>
  <input
    name="file"
    onChange={[Function]}
    type="file"
  />
  <Component />
  <Component
    margin="10px"
  >
    Clique para adicionar uma m√≠dia
  </Component>
</styled.label>
`;
